source/TestData.cpp

```diff
@@ -0,0 +1,110 @@
+/* Test.cpp
+Copyright (c) 2019 by Peter van der Meer
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "DataFile.h"
+#include "DataNode.h"
+#include "DataWriter.h"
+#include "Files.h"
+#include "TestData.h"
+#include <string>
+
+using namespace std;
+
+
+
+TestData::TestData()
+{
+	dataSetName = "";
+	dataSetType = 0;
+	sourceDataFile = "";
+}	
+
+
+
+TestData::~TestData()
+{
+}
+
+
+
+string TestData::Name()
+{
+	return dataSetName;
+}
+
+
+
+// Loader to load the generic test-data entry
+void TestData::Load(const DataNode &node, const string sourceDataFilePath)
+{
+	sourceDataFile = sourceDataFilePath;
+	if(node.Size() < 2)
+	{
+		node.PrintTrace("No name specified for test-data dataset");
+		return;
+	}
+	if (node.Token(0) != "test-data")
+	{
+		node.PrintTrace("Non-test found in test-data dataset parsing");
+		return;
+	}
+	dataSetName = node.Token(1);
+	
+	for(const DataNode &child : node)
+		// Only need to parse the category for now. The contents will be
+		// scanned for at write-out of the test-data.
+		if (child.Size() > 1 && child.Token(0) == "category" && child.Token(1) == "savegame")
+			dataSetType = TestData::SAVEGAME;
+}
+
+
+
+// Inject the test-data to the proper location
+bool TestData::Inject() const
+{
+	// We only know how to inject savegame data for now.
+	if ((dataSetType != TestData::SAVEGAME) || (dataSetName.empty()) || (sourceDataFile.empty()))
+		return false;
+
+	// Open the source-file and scan until we find the test-data
+	// Then scan for the contents keyword
+	// Then write out the complete contents to the target file
+	DataFile sourceData(sourceDataFile);
```

Comment:
Use the `{ }` syntax where possible over `( )`

Reply:
I'm not sure what you mean here, but I'll check if there is a constructor type that uses an `{}` syntax.

Reply:
Done